### PR TITLE
Add performance testing blog post

### PR DESCRIPTION
**Pull request by martinkuba**
**Preview**: https://deploy-preview-3470--opentelemetry.netlify.app/blog/2023/perf-testing/

### CODE REVIEW

File: content/en/blog/2023/perf-testing/index.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+A number of OpenTelemetry SDKs already include existing micro-benchmark tests.
+These tests measure performance of smaller pieces of the OpenTelemetry SDK in
+isolation.
+
+Example benchmark tests:
+
+- [Java](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/jmh/java/io/opentelemetry/sdk/trace/SpanBenchmark.java)
+- [Python](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/tests/performance/benchmarks/trace/test_benchmark_trace.py)
+- [Go](https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/trace/benchmark_test.go)
+- [JavaScript](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/test/performance/benchmark/span.js)
```
Comment (chalin) [inline content/en/blog/2023/perf-testing/index.md:59]:
```suggestion
A number of OpenTelemetry SDKs already include existing micro-benchmark tests,
for example:

- [SpanBenchmark.java](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/jmh/java/io/opentelemetry/sdk/trace/SpanBenchmark.java)
- [test_benchmark_trace.py](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/tests/performance/benchmarks/trace/test_benchmark_trace.py)
- [benchmark_test.go](https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/trace/benchmark_test.go)
- [benchmark/span.js](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/test/performance/benchmark/span.js)
```

File: content/en/blog/2023/perf-testing/index.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+The tests are also run on community-owned bare metal machines, so that the
+results are as consistent as possible.
+
+Published
+[benchmark results for Java](https://open-telemetry.github.io/opentelemetry-java/benchmarks/)
+![Java benchmark results](java-benchmark-results.png 'Java benchmark results')
+
+Published
+[benchmark results for JavaScript](https://open-telemetry.github.io/opentelemetry-js/benchmarks/)
+![JavaScript benchmark results](js-benchmark-results.png 'JavaScript benchmark results')
```
Comment (chalin) [inline content/en/blog/2023/perf-testing/index.md:73]: Using `figure` renders the images better:

```suggestion
{{% figure
  src="java-benchmark-results.png"
  caption="Sample [benchmark results for Java](https://open-telemetry.github.io/opentelemetry-java/benchmarks/)"
%}}

{{% figure
  src="js-benchmark-results.png"
  caption="Sample [benchmark results for JavaScript](https://open-telemetry.github.io/opentelemetry-js/benchmarks/)"
%}}
```

File: content/en/blog/2023/perf-testing/index.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+date: 2023-11-01
+author: '[Martin Kuba](https://github.com/martinkuba) (Lightstep)'
+cSpell:ignore: Kuba
+---
+
+As more and more users are looking to use OpenTelemetry instrumentation in their
+production deployments, one important consideration is the impact that
+OpenTelemetry will have on their application performance. In this blog post I
+will discuss a few recent improvements in tooling around performance
+benchmarking. Measuring performance overhead Instrumentation is not free - it
```
Comment (chalin) [inline content/en/blog/2023/perf-testing/index.md:13]:
```suggestion
benchmarking. Measuring performance overhead Instrumentation is not free, it
```

File: content/en/blog/2023/perf-testing/index.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+intercepts an application's operations and collects (often) a large amount of
+data, which takes additional CPU and memory. This can have a direct effect on
+throughput and response time, which can affect the end-user experience with the
+application. It can also have an impact on operational cost, e.g. increasing the
+number of instances a service runs on.
+
+Providing general guidance about performance overhead is inherently difficult.
+There are many factors that affect performance - the application throughput,
+hardware the application runs on, what exactly is instrumented, how the
+OpenTelemetry SDK is configured, sampling etc. Ultimately the best way to
```
Comment (chalin) [inline content/en/blog/2023/perf-testing/index.md:23]:
```suggestion
OpenTelemetry SDK is configured, sampling, etc. Ultimately, the best way to
```

**Review by chalin — COMMENTED** [review]
Nice!

See inline for copyedits.

**Comment by martinkuba** [comment]
@chalin Thank you for the review! I have made the changes based on your suggestions.

**Review by cartersocha — APPROVED** [review]
lgtm

**Comment by chalin** [comment]
> The TEXT linter is failing because it wants to capitalize `java` in SpanBenchmark.java

I've submitted a fix (#3492). We'll need to wait for that to land so that you can rebase.

> REFCACHE check, I tried running `npm run test-and-fix`, but it's not updating anything

Hmm, `test-and-fix` works for me. I've submitted the updated `refcache` file.

**Comment by cartermp** [comment]
@cartersocha agreed, this can be post-kubecon. E_TOO_MANY_BLOG_POSTS

**Comment by chalin** [comment]
All checks are ✅

File: content/en/blog/2023/perf-testing/index.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+measure performance is in the context of the specific application by running a
+load test.
+
+With that said a number of OpenTelemetry components include performance tests
+that help catch regressions and can be used to provide some idea of their
+performance characteristics.
+
+### OpenTelemetry Collector
+
+The OpenTelemetry Collector runs
```
Comment (svrnm) [inline content/en/blog/2023/perf-testing/index.md:36]:
```suggestion
The [OpenTelemetry Collector](/docs/collector/) runs
```

**Review by svrnm — APPROVED** [review]
overall LGTM, but I agree let's push this out post KubeCon

File: content/en/blog/2023/perf-testing/index.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
+---
+title: OTel component performance benchmarks
+linkTitle: Performance benchmarks
+date: 2023-11-01
```
Comment (svrnm) [inline content/en/blog/2023/perf-testing/index.md:4]: Are we good to publish this one? Let's set a date!

**Review by chalin — APPROVED** [review]

File: content/en/blog/2023/perf-testing/index.md (hunk 6 above)
Comment (cartermp) [inline content/en/blog/2023/perf-testing/index.md:4]: Let's go with next Monday
```suggestion
date: 2023-11-27
```

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
Hrm. Did https://github.com/open-telemetry/opentelemetry.io/pull/3492 not fix the issue, or is there just some git branch fuckery going on causing the textlint check to fail again?

**Comment by svrnm** [comment]
> Hrm. Did #3492 not fix the issue, or is there just some git branch fuckery going on causing the textlint check to fail again?

Seems that the changes from that PR got not carried over when we switched from JSON to YAML

**Review by JaredTan95 — APPROVED** [review]

**Comment by cartermp** [comment]
Will be published on Monday.

**Pull request merged**